lib/io.js

```diff
@@ -423,11 +386,7 @@ DatumReader.prototype = {
             var readersField = readersSchema.fieldsHash[field.name];
             if (readersField) {
                 var data = self.readData(field.type, readersField.type, decoder);
-                if (data instanceof AvroErrors.BlockDelayReadError) {
-                    decoder._input.skip(oldOffset - decoder._input.offset);
-                    return data;
-                } else
-                    record[field.name] = data;
+                record[field.name] = data;
             } else {
                 console.error('SKIPPING');
```

Comment:
I'll look at that later and will consider it as part of a separate diff.  I don't want to include that here.